### ISSUE TITLE
Support for local subs with `my sub`

### ISSUE THREAD

**Issue by EvanCarroll** [Perl-Critic/Perl-Critic#946]
I'm getting

> [error:PerlCritic] [Subroutines::ProhibitNestedSubs] MyApp/DNS.pm: Nested named subroutine at line 259, column 5

But I don't feel like `ProhibitNestedSubs` should exclude `my sub` which are explicitly supposed to be nested.

**Comment by nanto** [comment]
Responding to https://github.com/Perl-Critic/Perl-Critic/pull/971#issuecomment-981706268, I think we can find nested sub by inspecting its ancestor nodes, instead of inspecting its descendant nodes. Inspecting ancestor nodes also resolves #972.

I tried the following code on my local branch and that code passes test cases added in #971:

```perl
package Perl::Critic::Policy::Subroutines::ProhibitNestedSubs;

sub violates {
    my ($self, $elem, $doc) = @_;

    return if $elem->isa('PPI::Statement::Scheduled') || defined $elem->type;

    my $outer = $elem;
    while ($outer = $outer->parent) {
        last if $outer->isa('PPI::Statement::Sub')
            &&  ! $outer->isa('PPI::Statement::Scheduled');
    }
    return if not $outer;

    # Must be a violation...
    return $self->violation($DESC, $EXPL, $elem);
}
```

**Comment by petdance** [comment]
I'm not too concerned with implementation.  I just want to make sure we have tested the corner cases.
